File: examples/swift/constructs/functions_basic.swift
// Ported from examples/lumen/constructs/functions_basic.lm by scripts/port_examples.py; edit the Lumen original, not this file.
func square(x: Int) -> Int {
    return x * x
}

func add(a: Int, b: Int) -> Int {
    return a + b
}

func greet(name: String) -> String {
    return "Hello, " + name
}

print("Test: Basic Functions")
print(square(x: 5))
print(add(a: 10, b: 20))
print(greet(name: "Lumen"))
func get_constant() -> Int {
    return 42
}

print(get_constant())
func compute(x: Int, y: Int) -> Int {
    let sum = x + y
    let product = x * y
    return sum + product
}

print(compute(x: 3, y: 4))

File: examples/ruby/constructs/functions_basic.rb
# Ported from examples/lumen/constructs/functions_basic.lm by scripts/port_examples.py; edit the Lumen original, not this file.
def square(x)
    return x * x
end

def add(a, b)
    return a + b
end

def greet(name)
    return "Hello, " + name
end

puts("Test: Basic Functions")
puts(square(5))
puts(add(10, 20))
puts(greet("Lumen"))
def get_constant()
    return 42
end

puts(get_constant())
def compute(x, y)
    sum = x + y
    product = x * y
    return sum + product
end

puts(compute(3, 4))
